src/libraries/System.Private.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Runtime.cs

```diff
@@ -244,6 +246,18 @@ public static int GetJSObjectId(object rawObj)
                 js.GetWrappedObject() ?? h.Target : h.Target;
         }
 
+        public static object? GetDotNetObjectInFlight(int gcHandle)
```

Comment:
I don't fully grasp what this method does, especially the scenario where the target isn't a JSObject. It would be good to add a doc comment explaining how it is intended to be used.

Reply:
I added _bool-as-int_ parameter `shouldAddInflight` into the pre-existing GetDotNetObject instaed of new method, because it's still same functionality and it's more readabe. Also wrote explanation comment.